src/fu-security-attr.c

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (C) 2020 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include <config.h>
+#include <glib/gi18n.h>
+
+#include "fu-security-attr.h"
+
+const gchar *
+fu_security_attr_get_name (FwupdSecurityAttr *attr)
+{
+	const gchar *appstream_id = fwupd_security_attr_get_appstream_id (attr);
+	if (g_strcmp0 (appstream_id, FWUPD_SECURITY_ATTR_ID_SPI_BIOSWE) == 0) {
+		/* TRANSLATORS: Title: SPI refers to the flash chip in the computer */
+		return _("SPI write");
+	}
+	if (g_strcmp0 (appstream_id, FWUPD_SECURITY_ATTR_ID_SPI_BLE) == 0) {
+		/* TRANSLATORS: Title: SPI refers to the flash chip in the computer */
+		return _("SPI lock");
+	}
+	if (g_strcmp0 (appstream_id, FWUPD_SECURITY_ATTR_ID_SPI_SMM_BWP) == 0) {
+		/* TRANSLATORS: Title: SPI refers to the flash chip in the computer */
+		return _("BIOS region of SPI");
+	}
+	if (g_strcmp0 (appstream_id, FWUPD_SECURITY_ATTR_ID_ACPI_DMAR) == 0) {
+		/* TRANSLATORS: Title: DMA as in https://en.wikipedia.org/wiki/DMA_attack  */
+		return _("Pre-boot kernel DMA protection");
+	}
+	if (g_strcmp0 (appstream_id, FWUPD_SECURITY_ATTR_ID_INTEL_AMT) == 0) {
+		/* TRANSLATORS: Title: AMT = Active Management Technology */
+		return _("Intel AMT");
+	}
+	if (g_strcmp0 (appstream_id, FWUPD_SECURITY_ATTR_ID_INTEL_CET) == 0) {
+		/* TRANSLATORS: Title: CET = Control-flow Enforcement Technology */
+		return _("Intel CET");
+	}
+	if (g_strcmp0 (appstream_id, FWUPD_SECURITY_ATTR_ID_INTEL_TME) == 0) {
+		/* TRANSLATORS: Title: TME = Total Memory Encryption */
+		return _("Intel TME");
```

Comment:
Although it's intel specific right now the way we implemented, can we leave intel out of this?  AMD and others should have their own and I think it's the same way we should show it.

Reply:
`s\FWUPD_SECURITY_ATTR_ID_INTEL_TME\FWUPD_SECURITY_ATTR_ID_INTEL_ENCRYPTED_MEMORY` ? Then we have `Supported` and `Unsupported.

Reply:
Well i was hoping to make it so that we don't have intel at all in the attribute and name.   AMD has SME and SEV, and I think the same attribute should be shared for them too, no?

Reply:
Gahh, that was supposed to be `FWUPD_SECURITY_ATTR_ID_ENCRYPTED_MEMORY` sorry

Reply:
Maybe memory isn't specific enough. `_DRAM`?

Reply:
Yeah that's good